tests/test_type_caster.py

```diff
@@ -1,4 +1,54 @@
+import pytest
 
+from data_utils.file_utils.fixed_width.type_caster import TypeCaster
 
-def test_cast_data_types():
-    assert 1 == 1
+
+def test_none_cast_data_types():
+    """if value is None the func should return None"""
+    result = TypeCaster.cast_data_types(None, value=None, data_type="", pic_clause="", field_name="")
+    assert result is None
+
+
+def test_parse_decimal_field_random():
+    """should return the original value with "{" removed if non decimal value is passed """
+    value = 'fail4sure{'
+    result = TypeCaster._parse_decimal_field(value=value, field_name='TRANSACTION_AMOUNT', pic_clause='-9(7).9(2)')
+    assert result == 'fail4sure'
+
+
+def test_parse_decimal_field_decimal_value():
+    """if input str is a float value then func should return float"""
+    value = '-0000123.45'
+    result = TypeCaster._parse_decimal_field(value=value, field_name='TRANSACTION_AMOUNT', pic_clause='-9(7).9(2)')
+    assert result == -0000123.45
+
+
+def test_integer_val_parse_decimal_field():
+    """if input is integer value should cast into decimal value based on pic clause"""
+    value = '-000012345'
+    result = TypeCaster._parse_decimal_field(value=value, field_name='TRANSACTION_AMOUNT', pic_clause='-9(7).9(2)')
+    assert result == -0000123.45
+
+
+def test_parse_time_field():
+    """string should be sliced to time format hh:mm:ss"""
+    parse_time_field: str = lambda x: TypeCaster._parse_time_field(value=x)
```

Comment:
You can leave this but in general it seems a bit more verbose to do it this way since you are just changing the name of the function. All you are saving is the 12 characters from `TypeCaster._` and makes the next person have to figure out why you are using a lambda here.

Reply:
makes sense I will change it